File: week-1/debugging_fizzbuzz/exercise_1/lib/fizzbuzz.rb
class FizzBuzz
  def play(number)
    string = ""
    string << "Fizz" if is_divisible_by?(3, the_number)
    string << "Bizz" if is_divisible_by?(5, number)
    string.empty? ? string : number
  end

  private

  def is_divisible_by?(divis0r, number)
    number % divisor == 1
  end
end

File: week-5/following_the_flow_and_getting_visibility_in_javascript_es6/fizzbuzz.js
class FizzBuzz {
  play(number) {
    if (this._isDivisibleBy(15, number)) {
      return 'FizzBuzz';
    } else if (this._isDivisibleBy(3, number)) {
      return 'Fizz';
    } else if (this._isDivisibleBy(5, number)) {
      return 'Buzz';
    } else {
      return number;
    }
  }

  _isDivisibleBy(divisor, number) {
    return number % divisor === 0;
  }
}

var fizzBuzz = new FizzBuzz();

for (let i = 1; i <= 100; i++) {
  console.log(fizzBuzz.play(i));
}
